search.go

```diff
@@ -96,6 +110,20 @@ type SearchResponse struct {
 
 	// Error if OpenSearch failed but responded with errors
 	Error *Error
```

Comment:
Yeah, in this case a pointer is correct so that the value can be treated like an `error` (i.e. `if r.Error != nil {`) without requiring consumers to type assert to get further details.